### PR TITLE
DM-3704: Refactor ImageDifferenceTask

### PR DESCRIPTION
**Pull request by djreiss**
Fix default setting
Move registerImage.py to ip_diffim
- Also move testRegister.py
Add refactored copy of imageDifference for testing
Fix imports, configs, schemas
Small fixes; refactor evaluate metrics
Confirmed that Zogy option works in refactor
Add individual command-line tasks
Fix usage of selectSources between tasks

### CODE REVIEW

File: bin.src/makeDiffim.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008, 2009, 2010 LSST Corporation.
```
Comment (parejkoj) [inline bin.src/makeDiffim.py]: Update copyright to include 2017.

File: bin.src/processDiffim.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008, 2009, 2010 LSST Corporation.
```
Comment (parejkoj) [inline bin.src/processDiffim.py]: update copyright.

File: python/lsst/pipe/tasks/imageDifference.py [hunk 3]
```diff
@@ -33,20 +33,26 @@
...
 import lsst.ip.diffim.utils as diUtils
 
 FwhmPerSigma = 2 * math.sqrt(2 * math.log(2))
 IqrToSigma = 0.741
 
+subtractAlgorithmRegistry = pexConfig.makeRegistry(
+    doc="A registry of subtraction algorithms for use as a subtask in imageDifference",
+)
+
+subtractAlgorithmRegistry.register('al', ImagePsfMatchTask)
```
Comment (parejkoj) [inline python/lsst/pipe/tasks/imageDifference.py]: Should we call this "AlardLupton"? A two-letter acronym is rather short.

**Comment by jdswinbank** [comment]
Marked as “won't fix” in Jira.

**Pull request closed without merging**